subsys/pm/device_runtime.c

```diff
@@ -63,7 +63,7 @@ static int runtime_suspend(const struct device *dev, bool async)
 		(void)k_work_schedule(&pm->work, K_NO_WAIT);
 	} else {
 		/* suspend now */
-		ret = pm->action_cb(pm->dev, PM_DEVICE_ACTION_SUSPEND);
```

Comment:
I think we should not use `pm_device_state_set` for multiple reasons:

- It uses states, not actions (`pm_device_state_set` is still available for legacy reasons really, so let's not use in new code unless there is a good reason or no other way)
- `pm_device_state_set` does checks already done by the runtime PM

Note that I see another issue: now devices using runtime PM are ignored by system PM, but, domains will not be ignored by system. If system then suspends a domain, what will happen to a devices attached to the domain using runtime PM?

Reply:
The first one is something that makes sense and has to be addressed. Will add a commit doing it and ensure that the action function is used.
The second reason,  I need to think, I don't think we should have the same logic scattered across the code.

Reply:
Regarding device runtime power management, I think it is hard to automate anything here, I believe if the application decided to enable runtime pm for a device it is its responsibility do it for the the domain if it is worried about system pm or to  set a constraint in the system to not sleep and not suspend devices. 

Automatically enable runtime power management on domain can be problematic because the domain may already being used by another device.

What we can do is automatically get / put the domain according with the device usage, since per conception if a device is under the domain it needs the domain active to be active and that is what it does.